File: services/changeowl-web/src/types/artifact.ts
import { ArtifactIntelligence } from "./artifact-intelligence";

export interface Artifact {
  id: string;
  title: string;
  body: string;
  url: string;
  author: string;
  createdAt: string;
  state?: string;

  repoName: string;
  repoId: string;

  techName: string;
  techId: string;

  intelligence?: ArtifactIntelligence;

  similarityScore?: number;
}

File: services/agent-service/app/models/artifact.py
from pydantic import BaseModel

from app.models.artifact_intelligence import ArtifactIntelligence


class Artifact(BaseModel):
    id: int

    title: str
    body: str | None = None
    url: str
    author: str

    createdAt: str
    state: str | None = None

    repoName: str
    repoId: int

    techName: str
    techId: int

    intelligence: ArtifactIntelligence | None = None

    similarityScore: float | None = None